resource/stack/stack.js

```diff
@@ -0,0 +1,141 @@
+const _ = require( "lodash" );
+const Promise = require("bluebird");
+const rancherFn = require( "../../src/rancher" );
+const format = require( "util" ).format;
+const environment = require( "../../src/data/nedb/environment" );
+const rp = require( "request-promise" );
+
+function sendMessage(slack, channels, message) {
+	channels.forEach( channel => {
+		slack.send( channel, message );
+	}, this);
+}
+
+function shouldUpgradeStack ( stack, catalog, branch, version ) {
+	if (stack.externalId) {
+		const stackInfo = stack.externalId.split("://")[1].split(":");
+		const stackCatalog = stackInfo[0];
+		const stackBranch = stackInfo[1];
+		const stackVersion = stackInfo[2];
+		return stackCatalog === catalog &&
+			stackBranch === branch &&
+			parseInt(stackVersion) < parseInt(version);
+	} else {
+		return false;
+	}
+}
+
+const getTemplate = Promise.coroutine(function* ( token, catalogOwner, catalog, branch, version ) {
+
+	let response =  yield rp( format( "https://api.github.com/repos/%s/%s/contents/templates/%s/%s", catalogOwner, catalog, branch, version ), {
+		qs: {
+			access_token: token, // eslint-disable-line
+			ref: "master"
+		},
+		headers: {
+			"User-Agent": "cowpoke"
+		},
+		json: true
+	} );
+	
+	const templateResult = {version};
+
+	for ( var i = 0; i < response.length; i++ ) {
+		templateResult[response[i].name] = yield rp( response[i].download_url, {
+			qs: {
+				access_token: token // eslint-disable-line
+			},
+			headers: {
+				"User-Agent": "cowpoke"
+			}
+		} );
+	}
+
+	return templateResult;
+	
+});
+
+const upgradeStack = Promise.coroutine(function* ( slack, envelope ) {
+
+	//read args
+	const githubInfo = envelope.data.catalog ? envelope.data.catalog.split("/") : [];
+	const githubOwner = githubInfo[0];
+	const githubRepo = githubInfo[1];
+	const rancherCatalogName = envelope.data.rancher_catalog_name;
```

Comment:
the naming scheme for the data being posting breaks convention from the other endpoints (camelCased properties vs _ delimited properties)